src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/ListPackageCommandRunner.cs

```diff
@@ -178,6 +321,18 @@ private static void WarnForHttpSources(ListPackageArgs listPackageArgs, ListPack
                 }
             }
 
+            foreach (PackageSource packageSource in auditSources)
+            {
+                if (packageSource.IsHttp && !packageSource.IsHttps && !packageSource.AllowInsecureConnections)
+                {
+                    if (httpPackageSources == null)
+                    {
+                        httpPackageSources = new();
+                    }
+                    httpPackageSources.Add(packageSource);
+                }
+            }
```

Comment:
The majority of this code is duplicated from the previous for loop. We can add a helper method to reduce the code duplication.